src/handle.rs

```diff
@@ -45,15 +45,19 @@ pub fn find_next(
 
 pub async fn collect_data(info: Info) -> anyhow::Result<Data> {
     let mut clients = Clients::get_async().await?
+        .iter()
         .filter(|c| c.workspace.id != -1) // ignore nonexistent clients
         .filter(|w| !info.hide_special_workspaces || !w.workspace.id < 0)
+        .cloned()
         .collect::<Vec<_>>();
 
     let monitors = Monitors::get_async().await?;
 
     // get all workspaces sorted by ID
+    let workspaces = Workspaces::get_async().await?;
     let workspaces = {
-        let mut workspaces = Workspaces::get_async().await?
+        let mut workspaces = workspaces
```

Comment:
same here, just add ``.into_iter()`` after the ``let mut workspaces = Workspaces::get_async().await?`` call:

```rust
// get all workspaces sorted by ID
let workspaces = {
    let mut workspaces = Workspaces::get_async().await?
        .into_iter()
        .filter(|w| w.id != -1) // ignore nonexistent clients
        .filter(|w| !info.hide_special_workspaces || !w.id < 0)
        .collect::<Vec<_>>();

    workspaces.sort_by(|a, b| a.id.cmp(&b.id));
    workspaces
};
```